src/actions/FioActions.js

```diff
@@ -51,22 +55,47 @@ export const refreshConnectedWallets = async (dispatch: Dispatch, getState: GetS
 }
 
 export const checkFioObtData = (walletId: string, transactions: EdgeTransaction[]) => async (dispatch: Dispatch, getState: GetState) => {
+  if (checkFioObtDataCounter > MAX_OBT_DATA_CHECKS) return // to prevent loop if account does not have fio wallets
   const state = getState()
   const { account } = state.core
-  if (!account || !account.currencyConfig) {
-    setTimeout(() => {
+  const fioWallets = state.ui.wallets.fioWallets
+  if (account == null || account.currencyConfig == null || fioWallets.length === 0) {
+    checkFioObtDataCounter++
+    return setTimeout(() => {
       dispatch(checkFioObtData(walletId, transactions))
     }, 400)
   }
   try {
     const fioPlugin = account.currencyConfig[CURRENCY_PLUGIN_NAMES.FIO]
+    const wallet = account.currencyWallets[walletId]
+
+    const obtDataRecords = await getFioObtData(fioWallets)
 
     for (const transaction: EdgeTransaction of transactions) {
-      if (transaction.metadata) {
-        const { name } = transaction.metadata
+      const edgeMetadata: EdgeMetadata = transaction.metadata != null ? transaction.metadata : { notes: '' }
+      try {
+        const { name } = edgeMetadata
         if (name && (await fioPlugin.otherMethods.isFioAddressValid(name))) {
           addToFioAddressCache(state.core.account, [name])
         }
+      } catch (e) {
+        //
+      }
+      const obtForTx: FioObtRecord | void = obtDataRecords.find(obtRecord => obtRecord.content.obt_id === transaction.txid)
+      if (obtForTx == null) return
+
+      if (edgeMetadata.notes == null) edgeMetadata.notes = ''
+      let fioNotes = `${s.strings.fragment_transaction_list_sent_prefix}${s.strings.word_to_in_convert_from_to_string} ${obtForTx.payee_fio_address}`
+      if (obtForTx.content.memo != null) fioNotes += `\n${s.strings.fio_sender_memo_label}: ${obtForTx.content.memo}`
```

Comment:
Also check that the memo isn't an empty string before adding this line to the notes.